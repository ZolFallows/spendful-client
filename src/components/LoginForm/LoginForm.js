import React from 'react';
import AuthApiService from '../../services/auth-api-service';

class LoginForm extends React.Component {

  onSubmit = (ev) => {
    ev.preventDefault();

    const { email_address, password } = ev.target;

    AuthApiService.postLogin({
      email_address: email_address.value,
      password: password.value,
    })
    .then(res => {
      email_address.value = ''
      password.value = ''

      this.props.onSuccess(res.token);
    })
    .catch(err => {
      this.props.onFailure(err.errors)
    })
  }

  render() {

    return (
      <form onSubmit={this.onSubmit}>
        <input type="text" placeholder="Email" name="email_address" autoComplete="off" required/>
        <input type="password" placeholder="Password" name="password" required/>
        <button type="submit">Log In</button>
      </form>
    );
  }
}

export default LoginForm;
