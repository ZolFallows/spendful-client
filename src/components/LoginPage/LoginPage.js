import React from 'react';
import { Link } from 'react-router-dom';
import LoginForm from '../LoginForm/LoginForm';
import UserContext from '../../contexts/UserContext';

import './LoginPage.css';

class LoginPage extends React.Component {

  static contextType = UserContext;

  static defaultProps = {
    location:{},
    history:{
      push: () => { },
    },
  }

  constructor(props) {
    super(props);

    this.state = {
      errors: [],
    };

    this.handleLoginSuccess = this.handleLoginSuccess.bind(this);
    this.handleLoginFailure = this.handleLoginFailure.bind(this);
  }


  handleLoginSuccess = (token) => {
    const { location, history } = this.props
    const destination = (location.state || {}).from || '/dashboard'

    this.context.processLogin(token)
    history.push(destination);
  }

  handleLoginFailure = (errors) => {
    this.setState({errors});
  }

  render () {

    return(
      <section className="login-form">

        {this.state.errors.length > 0 ? <div className="alert-error">{this.state.errors}</div> : ''}

        <LoginForm onSuccess={this.handleLoginSuccess} onFailure={this.handleLoginFailure} />

        <Link to="/">Don't have an account?</Link>

      </section>
    )
  }
}

export default LoginPage;
