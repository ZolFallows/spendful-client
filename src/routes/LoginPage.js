import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import './LoginPage.css';

export default class LoginPage extends Component {
  static defaultProps = {
    location:{},
    history:{
      push: () => { },
    },
  }

  handleLoginSuccess = () => {
    const { location, history } = this.props
    const destination = (location.state || {}).from || '/dashboard'
    history.push(destination); 
  }

  render() {
    return(
      <section className="login-form">
        <form>
          <input type="text" placeholder="Username"/>
          <input type="password" placeholder="Password"/>
          <button type="submit">Log In</button> 
        </form>
        <Link to="/">Don't have an account?</Link>
      </section>
    )
  }
}
