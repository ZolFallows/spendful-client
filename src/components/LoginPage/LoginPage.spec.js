import React from 'react';
import ReactDOM from 'react-dom';
// import { Switch, Route, withRouter, Redirect } from 'react-router-dom';
import { BrowserRouter } from 'react-router-dom';
import Enzyme from 'enzyme'
import { shallow } from 'enzyme';
// import { expect } from 'chai';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import LoginPage from './LoginPage';
import LoginForm from '../LoginForm/LoginForm';

Enzyme.configure({ adapter: new Adapter() });


describe('Login Page component', () => {
     const testEmail = 'test@spendful.com';
     const testPass = 'Endgame';

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
         <BrowserRouter>
           <LoginPage />
         </BrowserRouter>,
          
          div);
        ReactDOM.unmountComponentAtNode(div);
      });

    

    it('renders a <LoginForm /> component', () => {

      let wrapper = shallow(<LoginForm />);

      expect(wrapper.find(LoginForm))
        

    });
    
    it('displays a email and password field for login', () => {

      let wrapper = shallow(<LoginForm />);

      expect(wrapper.find(`input[type="text"]`).simulate('change', {target: {name: 'email_address', value: testEmail}}));
      expect(wrapper.find(`input[type="text"]`).text()).toEqual('');

      //.simulate('change', {target: {name: 'email_address', value: testEmail}}));
      //expect(wrapper.state('email_address')).toEqual(testEmail);

      expect(wrapper.find('input[type="password"]').simulate('change', {target: {name: 'password', value: testPass}}));
      expect(wrapper.find(`input[type="password"]`).text()).toEqual('');

      expect(wrapper.find('button[type="submit"]').text()).toEqual('Log In');

      //.simulate('change', {target: {name: 'password', value: testPass}});
      //expect(wrapper.state('password')).toEqual(testPass);

      expect(wrapper).toMatchSnapshot();

    });

    it('renders successful login with right data', () => {

      let wrapper = shallow(<LoginForm />);

      wrapper.find(`input[type="text"]`).simulate('change', {target: {name: 'email_address', value: testEmail}});

      wrapper.find('input[type="password"]').simulate('change', {target: {name: 'password', value: testPass}});

      wrapper.find('button').simulate('click');

      wrapper = shallow(<div className="alert-error"></div>);

      expect(wrapper.find('.alert-error')).contains('');

      

      



    })

    it('displays error if no password is provided', () => {

      let wrapper = shallow(<LoginForm />);

      wrapper.find(`input[type="text"]`).simulate('change', {target: {name: 'email_address', value: testEmail}});

      wrapper.find('input[type="password"]').simulate('change', {target: {name: 'password', value: ''}});

      wrapper.find('button').simulate('click');

      wrapper = shallow(<div className="alert-error"></div>);

      expect(wrapper.find('.alert-error')).contains('Incorrect Email or Password');

    });

    it('display error if user provides an email that does exist in db', () => {

    });
    
    it('display error if user provides an incorrect password', () => {

    });
    
});
