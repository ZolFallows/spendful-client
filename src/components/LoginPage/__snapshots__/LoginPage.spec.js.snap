// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page component displays a email and password field for login 1`] = `ShallowWrapper {}`;
